main/og_to_journal.es6.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+global.FOAM_FLAGS = {gcloud: true};
+require('foam2');
+
+require('../lib/confluence/aggressive_removal.es6.js');
+require('../lib/confluence/api_velocity.es6.js');
+require('../lib/confluence/api_velocity_data.es6.js');
+require('../lib/confluence/browser_metric_data.es6.js');
+require('../lib/confluence/browser_specific.es6.js');
+require('../lib/confluence/failure_to_ship.es6.js');
+require('../lib/confluence/set_ops.es6.js');
+require('../lib/dao_container.es6.js');
+require('../lib/datastore/datastore_container.es6.js');
+require('../lib/web_apis/release.es6.js');
+require('../lib/web_apis/release_interface_relationship.es6.js');
+require('../lib/web_apis/web_interface.es6.js');
+require('../lib/web_catalog/object_graph_importer.es6.js');
+const pkg = org.chromium.apis.web;
+
+const logger = foam.log.ConsoleLogger.create();
+
+function getJournalDAO(name, cls, ctx, mode) {
+  const filename = path.resolve(__dirname, `../data/${name}-journal.js`);
+  logger.info(`Creating JDAO (mode=${mode}) in ${filename}`);
+  return foam.dao.JDAO.create({
+    of: cls,
+    delegate: foam.dao.MDAO.create({of: cls}, ctx),
+    journal: foam.dao.NodeFileJournal.create({
+      of: cls,
+      fd: fs.openSync(
```

Comment:
A bit of deja vu here and the filename bit above. Is there some code sharing that should be done here?

Reply:
Perhaps. I've been a little lazy about code sharing in `main` because we are likely to move to table-based (instead of join-based) model in our long term vision for performance improvements.

Reply:
That's OK, you're the maintainer and make the trade-off better than anyone else.